063-topic-replication-factor-change.md

```diff
@@ -0,0 +1,240 @@
+
+| Name                                   | Type    | Default | Description                                                   |
+|:---------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED         | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_NAMESPACE       | String  | ""      | Namespace where Cruise Control is running                     |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME        | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT            | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED     | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED    | Boolean | false   | Whether Basic authentication is enabled                       |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED    | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_SECRET_NAME     | String  | ""      | Secret name containing the truststore                         |
+| STRIMZI_CRUISE_CONTROL_API_SECRET_NAME | String  | ""      | Secret name containing the API credentials                    |
+
+In case of Cluster Operator deployment, a dedicated admin user is automatically created for the Topic Operator.
+
+In case of standalone deployment, the Cluster Operator expects the following secrets to be created by the user:
+
+- A secret with `cruise-control.p12` key containing the truststore in PKCS12 format, and `cruise-control.password` key containing the truststore password.
+- A secret with `cruise-control.apiToAdminUsername` key containing the API username with admin role, and `cruise-control.apiToAdminPassword` key containing the user password.
+
+#### Error handling
+
+During the whole replicas change process, the KafkaTopic resource remains in a Ready state.
+Any KafkaTopic with an ongoing replicas change has a status condition of type Ongoing and reason ReplicasChanged, and a new status field called  `userTaskId`, which is used to rebuild the cache on ReplicasChangeManager's initialization.
+Pending replicas changes that were not marked as ongoing (unknown to Cruise Control), are simply re-added to pendingChanges by the Topic Operator reconciliation logic.
+
+Ongoing replicas change status example:
+
+```sh
+status:
+  conditions:
+  - lastTransitionTime: "2024-01-12T14:54:02.833067402Z"
+    status: "True"
+    type: Ready
+  - lastTransitionTime: "2024-01-12T15:01:24.395861299Z"
+    message: Replicas change ongoing
+    reason: ReplicasChanged
+    status: "True"
+    type: Ongoing
+  observedGeneration: 7
+  topicName: my-topic
+  userTaskId: 750dfc95-c67b-431c-979b-43843ac177e7
```

Comment:
Should there be some subsection? Something like

```
status:
  # ...
  replicationFactorChange:
    userTaskId: ...
```

This would provide more clarity to what it is and make it easier to add more fields easily if needed. (As I said, there should probably be a separate section for API/CRD changes to discuss this).

Reply:
Ok, thanks.